tests/shared/src/main/scala-3/org/typelevel/literally/examples/literals.scala

```diff
@@ -28,18 +28,12 @@ object literals:
       ${PortLiteral('ctx, 'args)}
 
   object ShortStringLiteral extends Literally[ShortString]:
-    def validate(s: String): Option[String] =
-      if s.length <= ShortString.MaxLength then None 
-      else Some(s"ShortString must be <= ${ShortString.MaxLength} characters")
-
-    def build(s: String)(using Quotes) =
-      '{ShortString.unsafeFromString(${Expr(s)})}
+    def validate(s: String)(using Quotes) =
+      if s.length <= ShortString.MaxLength then Right('{ShortString.unsafeFromString(${Expr(s)})}) 
+      else Left(s"ShortString must be <= ${ShortString.MaxLength} characters")
 
   object PortLiteral extends Literally[Port]:
-    def validate(s: String): Option[String] =
+    def validate(s: String)(using Quotes) =
       Try(s.toInt).toOption.flatMap(Port.fromInt) match
```

Comment:
`toIntOption`?

Reply:
Ha, I didn't use that b/c this project cross compiles for 2.12 but this file is Scala 3 only so I'll update. :)